### PR TITLE
[CI-Examples] Add EDMM Support in manifest files

### PR DESCRIPTION
**Pull request by anjalirai-intel**
Add EDMM support in manifest files.
Increasing wait time for server because workloads takes 2x-5x times to execute with EDMM than non-EDMM.

Signed-off-by: Anjali Rai <anjali.rai@intel.com>

- - -
This change is [Reviewable](https://reviewable.io/reviews/gramineproject/examples/47)

### CODE REVIEW

**Review by dimakuv — CHANGES_REQUESTED** [review]
> Reviewed 10 of 10 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/47)* status: all files reviewed, 3 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @anjalirai-intel)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/examples/47#-NLWZb7J-pAagNojLuca):*
I would rephrase:
```
Add EDMM support to all examples

For each example, user must specify environment variable `EDMM=1`
during build to enable EDMM.

Signed-off-by: Anjali Rai <anjali.rai@intel.com>
```
___
*[`curl/Makefile` line 50 at r1](https://reviewable.io/reviews/gramineproject/examples/47#-NLW_TRG-8Aa4uXJBca-:-NLW_TRG-8Aa4uXJBca0:b-md1kba) ([raw file](https://github.com/gramineproject/examples/blob/d02ac21390e0fd02a45c47f566138910086697e4/curl/Makefile#L50)):*
> ```Makefile
> check: all
> 	(cd test-docroot; exec python3 -m http.server -b 127.0.0.1 19111) & httpd_pid=$$!; \
> 	../common_tools/wait_for_server 300 127.0.0.1 19111; \
> ```

Hm, wait, but this is not SGX-related. Here we're waiting for a native `python3 -m http.server` application to start. Please revert this change (and sorry that I misled you in a private conversation).
___
*[`openjdk/java.manifest.template` line 22 at r1](https://reviewable.io/reviews/gramineproject/examples/47#-NLW_kQR5a2druV4EPVZ:-NLW_kQR5a2druV4EPV_:b-o2zso8) ([raw file](https://github.com/gramineproject/examples/blob/d02ac21390e0fd02a45c47f566138910086697e4/openjdk/java.manifest.template#L22)):*
> ```
> sgx.thread_num = 64
> sgx.edmm_enable = {{ 'true' if env.get('EDMM', '0') == '1' else 'false' }}
> sgx.require_exinfo = {{ 'true' if env.get('EDMM', '0') == '1' else 'false' }}
> ```

Please add some explanation about `require_exinfo`:
```
...
sgx.thread_num = 64

# `require_exinfo = true` is needed because OpenJDK queries fault info on page faults
sgx.edmm_enable = {{ 'true' if env.get('EDMM', '0') == '1' else 'false' }}
sgx.require_exinfo = {{ 'true' if env.get('EDMM', '0') == '1' else 'false' }}

sgx.trusted_files = [
...
```

**Review by anjalirai-intel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/47)* status: 8 of 10 files reviewed, 3 unresolved discussions, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv)

___
*[`-- commits` line 2 at r1](https://reviewable.io/reviews/gramineproject/examples/47#-NLWZb7J-pAagNojLuca):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

I would rephrase:
```
Add EDMM support to all examples

For each example, user must specify environment variable `EDMM=1`
during build to enable EDMM.

Signed-off-by: Anjali Rai <anjali.rai@intel.com>
```
</blockquote></details>

Done.
___
*[`openjdk/java.manifest.template` line 22 at r1](https://reviewable.io/reviews/gramineproject/examples/47#-NLW_kQR5a2druV4EPVZ:-NLZpkcZ23VPPKgKGsSq:b-896fix) ([raw file](https://github.com/gramineproject/examples/blob/d02ac21390e0fd02a45c47f566138910086697e4/openjdk/java.manifest.template#L22)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Please add some explanation about `require_exinfo`:
```
...
sgx.thread_num = 64

# `require_exinfo = true` is needed because OpenJDK queries fault info on page faults
sgx.edmm_enable = {{ 'true' if env.get('EDMM', '0') == '1' else 'false' }}
sgx.require_exinfo = {{ 'true' if env.get('EDMM', '0') == '1' else 'false' }}

sgx.trusted_files = [
...
```
</blockquote></details>

Done.
___
*[`curl/Makefile` line 50 at r1](https://reviewable.io/reviews/gramineproject/examples/47#-NLW_TRG-8Aa4uXJBca-:-NLZpls-DfJ-nbmO4m8i:b-896fix) ([raw file](https://github.com/gramineproject/examples/blob/d02ac21390e0fd02a45c47f566138910086697e4/curl/Makefile#L50)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

Hm, wait, but this is not SGX-related. Here we're waiting for a native `python3 -m http.server` application to start. Please revert this change (and sorry that I misled you in a private conversation).
</blockquote></details>

Done.

**Review by dimakuv — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/47)* status: all files reviewed, 1 unresolved discussion, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @anjalirai-intel)

___
*[`openjdk/java.manifest.template` line 22 at r1](https://reviewable.io/reviews/gramineproject/examples/47#-NLW_kQR5a2druV4EPVZ:-NL_C8F919ke9XvxcRuP:b-rzithg) ([raw file](https://github.com/gramineproject/examples/blob/d02ac21390e0fd02a45c47f566138910086697e4/openjdk/java.manifest.template#L22)):*
<details><summary><i>Previously, anjalirai-intel wrote…</i></summary><blockquote>

Done.
</blockquote></details>

:( I explicitly wrote my proposed change in full, so that it's obvious where to put the comment (after `sgx.thread_num` and right-before `sgx.edmm_enable`). Please move the comment as I suggested. And add an empty line after `sgx.thread_num`.

**Review by anjalirai-intel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/47)* status: 9 of 10 files reviewed, 1 unresolved discussion, not enough approvals from maintainers (3 more required), not enough approvals from different teams (2 more required, approved so far: ) (waiting on @dimakuv)

___
*[`openjdk/java.manifest.template` line 22 at r1](https://reviewable.io/reviews/gramineproject/examples/47#-NLW_kQR5a2druV4EPVZ:-NL_KK5sDxSmy_nezdJb:br8i61b) ([raw file](https://github.com/gramineproject/examples/blob/d02ac21390e0fd02a45c47f566138910086697e4/openjdk/java.manifest.template#L22)):*
<details><summary><i>Previously, dimakuv (Dmitrii Kuvaiskii) wrote…</i></summary><blockquote>

:( I explicitly wrote my proposed change in full, so that it's obvious where to put the comment (after `sgx.thread_num` and right-before `sgx.edmm_enable`). Please move the comment as I suggested. And add an empty line after `sgx.thread_num`.
</blockquote></details>

Sorry, I did not realize that. I have made the appropriate changes.

**Review by dimakuv — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/gramineproject/examples/47)* status: all files reviewed, all discussions resolved, not enough approvals from maintainers (2 more required), not enough approvals from different teams (1 more required, approved so far: Intel)
